haystack/document_stores/memory.py

```diff
@@ -81,6 +101,11 @@ def __init__(
         self.duplicate_documents = duplicate_documents
         self.use_gpu = use_gpu
         self.scoring_batch_size = scoring_batch_size
+        self.use_bm25 = use_bm25
+        self.bm25_tokenization_regex = bm25_tokenization_regex
+        self.bm25_algorithm = bm25_algorithm
```

Comment:
I see that later this line is used for calls like `bm25_class = getattr(rank_bm25, self.bm25_algorithm)`. How about storing the class directly here?

```suggestion
        self.bm25_algorithm = getattr(rank_bm25, self.bm25_algorithm)
```

Reply:
I understand your point and my first implementation was similar to your proposal.
With the current implementation, if some user wanted to change the algorithm and call `update_bm25` manually, it would be feasible...

However I have no strong reason to prefer this solution, so I defer to you...